### DISCUSSION TITLE
How to add our custom dataset on qlora yaml

### DISCUSSION THREAD

**Discussion by charles-123456** [axolotl-ai-cloud/axolotl#1259] [Q&A] [answered]
Hi,
I need to set my own custom dataset  , But throwing lot more TypeError , Is that any reference available to set custom dataset ?
Eg:
I set 
datasets:
  - path: /content/gdrive/MyDrive/train_data.json
    
  I got 
  dataset_wrapper, dataset_prompter = get_dataset_wrapper(
  File "/content/axolotl/src/axolotl/utils/data.py", line 648, in get_dataset_wrapper
    if ":load_" in config_dataset.type:
TypeError: argument of type 'NoneType' is not iterable

**Comment by Tilemachoc** [comment]
You can upload your dataset to your HuggingFace 🤗 account and then add the information to the config.yml file:

datasets:
  - path: [Your huggingface name]/Demo-Dataset #Path to json dataset file in huggingface
    type:
    conversation:
dataset_prepared_path: [Your huggingface name]/Demo-Dataset #Path to json dataset file in huggingface
val_set_size: 0.05
output_dir: ./out

**Comment by NanoCode012** [comment, marked as answer]
Sorry for the unclear docs. You may use local datasets like so (as shown in readme):
```yaml
# local
datasets:
  - path: data.jsonl # or json
    ds_type: json # see other options below
    type: alpaca
```
